src/internal/base-component/index.ts

```diff
@@ -45,11 +46,14 @@ export function validateProps(
   excludedProps: Array<string>,
   allowedEnums: Record<string, Array<string>>
 ) {
+  if (!isDevelopment) {
+    return;
+  }
   for (const [prop, value] of Object.entries(props)) {
     if (excludedProps.includes(prop)) {
       throw new Error(`${componentName} does not support "${prop}" property when used in ${THEME} theme`);
     }
-    if (allowedEnums[prop] && !allowedEnums[prop].includes(value)) {
+    if (value && allowedEnums[prop] && !allowedEnums[prop].includes(value)) {
```

Comment:
This will also not throw if `value` is `false`. Is this intended?

Reply:
This function is currently used only for string enums, I do not expect a boolean there.

If something changes in the future, we can update the function